tests/unit_tests/test_source_mesh.py

```diff
@@ -0,0 +1,148 @@
+from itertools import product
+
+from pathlib import Path
+import pytest
+import numpy as np
+
+import openmc
+import openmc.lib
+
+from tests import cdtemp
+from tests.regression_tests import config
+from subprocess import call
+
+
+TETS_PER_VOXEL = 12
+
+# This test uses a geometry file with cells that match a regular mesh. Each cell
+# in the geometry corresponds to 12 tetrahedra in the unstructured mesh file.
+@pytest.fixture
+def model():
+    openmc.reset_auto_ids()
+
+    ### Materials ###
+    materials = openmc.Materials()
+
+    water_mat = openmc.Material(material_id=3, name="water")
+    water_mat.add_nuclide("H1", 2.0)
+    water_mat.add_nuclide("O16", 1.0)
+    water_mat.set_density("atom/b-cm", 0.07416)
+    materials.append(water_mat)
+
+    ### Geometry ###
+    # This test uses a geometry file that resembles a regular mesh.
+    # 12 tets are used to match each voxel in the geometry.
+
+    # create a regular mesh that matches the superimposed mesh
+    regular_mesh = openmc.RegularMesh(mesh_id=10)
+    regular_mesh.lower_left = (-10, -10, -10)
+    regular_mesh.dimension = (10, 10, 10)
```

Comment:
Same comments apply to this file as what appears in the corresponding regression test